lib/spack/spack/url.py

```diff
@@ -71,7 +71,7 @@ def find_list_url(url):
 
     url_types = [
         # e.g. https://github.com/llnl/callpath/archive/v1.0.1.tar.gz
-        (r'^(https://github.com/[^/]+/[^/]+)/archive/',
+        (r'(.*github\.com/[^/]+/[^/]+)/archive/',
```

Comment:
This change improved `spack versions` detection for a few packages that don't use `https`:

`apex`: http://github.com/khuck/xpress-apex/archive/v0.1.tar.gz
`cram`: http://github.com/llnl/cram/archive/v1.0.1.tar.gz
`openblas`: http://github.com/xianyi/OpenBLAS/archive/v0.2.15.tar.gz